strawberry/dataloader.py

```diff
@@ -45,10 +49,50 @@ def __len__(self) -> int:
         return len(self.tasks)
 
 
+class AbstractCache(Generic[K, T], ABC):
+    @abstractmethod
+    def get(self, key: K) -> Union[Future[T], None]:
+        pass
+
+    @abstractmethod
+    def set(self, key: K, value: Future[T]) -> None:
+        pass
+
+    @abstractmethod
+    def delete(self, key: K) -> None:
+        pass
+
+    @abstractmethod
+    def clear(self) -> None:
+        pass
+
+
+class DefaultCache(AbstractCache[K, T]):
+    def __init__(self, cache_key_fn=None):
```

Comment:
can we add a type for `cache_key_fn`?

maybe  Optional[Callable[[K], Hashable]]?

Reply:
Shoot I missed it during the merge 😣